services/121-service/src/payments/transactions/transactions.service.ts

```diff
@@ -86,8 +92,7 @@ export class TransactionsService {
     transaction.registration = registration;
     transaction.programFspConfigurationId =
       relationDetails.programFspConfigurationId;
-    transaction.program = program;
```

Comment:
So this filter is no longer needed because paymentId is now unique across programs already. I can imagine there might be more places where this comes back, but I did not actively see many cases so far. Did you happen to go over this explicitly?